networking_oneview/ml2/drivers/oneview/neutron_oneview_client.py

```diff
@@ -107,14 +109,14 @@ def create(self, session, network_dict):
                 network_type=network_type.capitalize(), seg_id=network_seg_id)
             oneview_network_id = common.id_from_uri(oneview_network.get('uri'))
             mappings = self._add_to_ligs(
-                network_type, physical_network, oneview_network)
-        elif mapping_type == common.FLAT_NET_MAPPINGS_TYPE:
-            oneview_network_id = self.flat_net_mappings.get(physical_network)
-        # BUG(nicodemos) This is not reachable. see: line 126
+                network_type, physical_network, oneview_network.get('uri'))
+            if not mappings:
+                self.oneview_client.ethernet_networks.delete(oneview_network)
+                return
```

Comment:
Shouldn't we raise an Exception here? Otherwise Neutron will not be able to do a rollback, the network will still exist in Neutron.

Reply:
In my tests the network still remained in OpenStack.